client/js/widgets/button.js

```diff
@@ -69,10 +69,20 @@ export class Button extends Widget {
       if(a.applyVariables) {
         if(Array.isArray(a.applyVariables)) {
           for(const v of a.applyVariables) {
-            if(v.parameter && v.variable)
-              a[v.parameter] = variables[v.variable];
-            else
-              problems.push('Entry in parameter applyVariables does not contain "parameter" and "variable".');
+            if(v.parameter && v.variable) {
+              a[v.parameter] = (v.index === undefined) ? variables[v.variable] : variables[v.variable][v.index];
+            } else if (v.parameter && v.template) {
+              for (let key in variables) {
+                a[v.parameter] = v.template.replace(/({([^}]+)})/g, function(i) {
+                  let key = i.replace(/{/, '').replace(/}/, '');
+                  return (variables[key] === undefined) ? "" : variables[key];
+                });
+              }
+            } else if (v.parameter && v.property) {
+              let w = isValidID(v.widget) ? widgets.get(v.widget) : this;
+              a[v.parameter] = (w.p(v.property) === undefined) ? null : w.p(v.property);
+            } else
```

Comment:
I usually add braces to all or none of the branches. Again: wouldn't have said anything if not for the `for` loop above.